cloudcix_primitives/hyperv.py

```diff
@@ -1,31 +1,858 @@
+              f' on Host {host}.',
+        3035: f'Failed to connect the Host {host} for the payload resize_primary_storage',
+        3036: f'Failed to resize the primary storage image to {size}GB on Host {host}',
+        3037: f'Failed to connect the Host {host} for the payload create_mount_dir',
+        3038: f'Failed to create mount dir {domain_path}{domain}\\mount on Host {host}',
+        3039: f'Failed to connect the Host {host} for the payload mount_primary_storage',
+        3040: f'Failed to mount primary storage on Host {host}',
+        3041: f'Failed to connect the Host {host} for the payload copy_unattend_file',
+        3042: f'Failed to copy unattend file to {domain_path}{domain}\\mount\\ on Host {host}',
+        3043: f'Failed to connect the Host {host} for the payload copy_network_file',
+        3044: f'Failed to copy network file to {domain_path}{domain}\\mount\\ on Host {host}',
+        3045: f'Failed to connect the Host {host} for the payload unmount_primary_storage',
+        3046: f'Failed to unmount primary storage at {domain_path}{domain}\\mount on Host {host}',
+        3047: f'Failed to connect the Host {host} for the payload delete_mount_dir',
+        3048: f'Failed to delete mount dir {domain_path}{domain}\\mount on Host {host}',
+        3049: f'Failed to connect the Host {host} for the payload create_domain',
+        3050: f'Failed to create domain {domain} on Host {host}',
+        3051: f'Failed to connect the Host {host} for the payload set_cpu',
+        3052: f'Failed to set cpu {cpu} to domain {domain} on Host {host}',
+        3053: f'Failed to connect the Host {host} for the payload set_ram',
+        3054: f'Failed to set ram {ram}MB to domain {domain} on Host {host}',
+        3055: f'Failed to connect the Host {host} for the remove_default_nic',
+        3056: f'Failed to remove default nic from domain {domain} on Host {host}',
+        3057: f'Failed to connect the Host {host} for the add_gateway_vlan',
+        3058: f'Failed to add gateway vlan {gateway_vlan} to domain {domain} on Host {host}',
+        3059: f'Failed to connect the Host {host} for the add_secondary_vlans',
+        3060: f'Failed to add secondary vlans to domain {domain} on Host {host}',
+        3061: f'Failed to connect the Host {host} for the add_secondary_storages',
+        3062: f'Failed to add secondary storages to domain {domain} on Host {host}',
+        3063: f'Failed to connect the Host {host} for the start_domain',
+        3064: f'Failed to start domain {domain} on Host {host}',
+    }
 
+    messages_list = []
 
-def scrubprep():
-    return(False, 'Not Implemted')
+    # validate primary_storage
+    def validate_primary_storage(ps, msg_index):
+        if ps is None:
+            messages_list.append(f'{messages[msg_index]}: {messages[msg_index]}')
+            return False
+        if type(primary_storage) is not str:
```

Comment:
`primary_storage` is undefined in this function

Reply:
Also as the payload is in an f string, it converts value to a string why is an error check needed here.